packages/sync-actions/src/product-types-actions.js

```diff
@@ -0,0 +1,208 @@
+import forEach from 'lodash.foreach'
+import flatten from 'lodash.flatten'
+import { deepEqual } from 'fast-equals'
+import createBuildArrayActions, {
+  ADD_ACTIONS,
+  REMOVE_ACTIONS,
+  CHANGE_ACTIONS,
+} from './utils/create-build-array-actions'
+import { buildBaseAttributesActions } from './utils/common-actions'
+import * as diffpatcher from './utils/diffpatcher'
+import extractMatchingPairs from './utils/extract-matching-pairs'
+
+const REGEX_NUMBER = new RegExp(/^\d+$/)
+const REGEX_UNDERSCORE_NUMBER = new RegExp(/^_\d+$/)
+const getIsChangedOperation = key => REGEX_NUMBER.test(key)
+const getIsRemovedOperation = key => REGEX_UNDERSCORE_NUMBER.test(key)
+
+export const baseActionsList = [
+  { action: 'changeName', key: 'name' },
+  { action: 'setKey', key: 'key' },
+  { action: 'setDescription', key: 'description' },
+]
+
+export function actionsMapBase(diff, previous, next) {
+  return buildBaseAttributesActions({
+    diff,
+    actions: baseActionsList,
+    oldObj: previous,
+    newObj: next,
+  })
+}
+
+const attributeDefinitionsActionsList = [
+  { action: 'changeLabel', key: 'label' },
+  { action: 'setInputTip', key: 'inputTip' },
+  { actionKey: 'newValue', action: 'changeInputHint', key: 'inputHint' },
+  { action: 'changeIsSearchable', key: 'isSearchable' },
+  {
+    actionKey: 'newValue',
+    action: 'changeAttributeConstraint',
+    key: 'attributeConstraint',
+  },
+]
+
+const getIsAnAttributeDefinitionBaseFieldChange = diff =>
+  diff.label ||
+  diff.inputHint ||
+  diff.inputTip ||
+  diff.attributeConstraint ||
+  diff.isSearchable
+
+function actionsMapEnums(attributeType, attributeDiff, previous, next) {
+  const getAddActionName = type =>
+    type === 'enum' ? 'addPlainEnumValue' : 'addLocalizedEnumValue'
+  const getChangeEnumOrderActionName = type =>
+    type === 'enum'
+      ? 'changePlainEnumValueOrder'
+      : 'changeLocalizedEnumValueOrder'
+  const getChangeEnumLabelActionName = type =>
+    type === 'enum'
+      ? 'changePlainEnumValueLabel'
+      : 'changeLocalizedEnumValueLabel'
+
+  const handler = createBuildArrayActions('values', {
```

Comment:
`buildArrayActions` instead of `handler` as it's what we `create*`?

Reply:
the convention has been `handler`, but yeah you are right